arch/arm64/common/sdfirm.lds.S

```diff
@@ -69,6 +69,13 @@ SECTIONS
 		__init_data_end = .;
 #endif
 		DATA_DATA
+
+#ifdef CONFIG_GEM5
+		. = ALIGN(PAGE_SIZE);
+		simpoint_heap_start = .;
+		*(.SIMPOINT_MEM_PAGES)
+		simpoint_heap_end = .;
+#endif
 	}
```

Comment:
You can define GEM5_HEAP macro in include/asm-generic/sdfirm.lds.h, say:
#ifdef CONFIG_GEM5
#define GEM5_HEAP \
<these lines>
#else
#define GEM5_HEAP
#endif

Then other architecture (say, riscv) can also invoke GEM5_HEAP.

Reply:
done